### DISCUSSION TITLE
auto extract ?

### DISCUSSION THREAD

**Discussion by olmopje** [zach-morris/plugin.program.iagl#299] [General]
is it possible to after download detect if file is a zip / rar and automagicly extract this ?

im asking because im extending the iagl with xemu,xenia,rpcs3   and making my own list for this asswell .

for the xbox 360 games allot of them are in a .rar format wich needs to be extracted first

ps: is there an easy way to generate a .dat file from internet archive automated or is it realy all manual labor haha its a bitch .

**Comment by zach-morris** [comment]
Extraction can be setup to it'll do it automatically for the entire game list pretty easily.
See [this](https://github.com/zach-morris/plugin.program.iagl/blob/main/resources/data/dat_files/Atari_5200_ZachMorris.xml#L17) as an example.
There's not an automated way from the addon.  I use python magic to make it happen.
